taraxa/state/dpos/precompiled/dpos_contract.go

```diff
@@ -1151,61 +1163,24 @@ func (self *Contract) state_put(key *common.Hash, state *State) {
 	}
 }
 
-func (self *Contract) apply_genesis_entry(validator_info *GenesisValidator) {
-	state, state_k := self.state_get(validator_info.Address[:], BlockToBytes(0))
-	if state != nil {
-		panic("apply_genesis_entry: state already exists")
-	}
-
-	if !self.validators.CheckValidatorOwner(&common.ZeroAddress, &validator_info.Address) {
-		panic("apply_genesis_entry: owner already exists")
-	}
-	state = new(State)
-	state.RewardsPer1Stake = bigutil.Big0
-
-	self.validators.CreateValidator(&validator_info.Owner, &validator_info.Address, 0, big.NewInt(0), validator_info.Commission, validator_info.Description, validator_info.Endpoint)
-	state.Count++
+func (self *Contract) apply_genesis_entry(validator_info *GenesisValidator, make_context func(caller *common.Address, value *big.Int) vm.CallFrame) {
+	args := RegisterValidatorArgs{}
+	args.FromGenesis(validator_info)
```

Comment:
wouldn't this be cleaner ? 
`args := validator_info.gen_register_validator_args() `